server/src/main/webapp/WEB-INF/rails/webpack/models/pipeline_configs/nameable_set.ts

```diff
@@ -24,6 +24,17 @@ export interface Nameable extends Validatable {
 // Specialized Set<T> implementation where member equality is based on only the `name`
 // of the material, and NOT the identity or the structure.
 export class NameableSet<T extends Nameable> extends ValidatableMixin implements Set<T> {
+  [Symbol.toStringTag]: string = `NameableSet`;
+  private readonly _members    = new Map<string, T>(); // preserves insertion order
+
+  constructor(items?: readonly T[] | null) {
```

Comment:
initialize items with an empty array?
`constructor(items?: readonly T[] = [])`

This will avoid the `if(items)` check in the code

Reply:
`items?: readonly T[]` is still an optional, hence it is needed to check undefined